src/login/connect_server.h

```diff
@@ -0,0 +1,2257 @@
+                if (ret == SQL_ERROR || sql->NumRows() == 0)
+                {
+                    ShowWarning(fmt::format("login_parse: user <{}> could not be found using the provided information. Aborting.", escaped_name));
+                    ref<uint8>(data_, 0) = LOGIN_ERROR;
+                    do_write(1);
+                    return;
+                }
+
+                ret = sql->NextRow();
+
+                uint32 accid  = sql->GetUIntData(0);
+                uint8  status = (uint8)sql->GetUIntData(1);
+
+                if (status & ACCOUNT_STATUS_CODE::BANNED)
+                {
+                    ShowInfo(fmt::format("login_parse: banned user <{}> detected. Aborting.", escaped_name));
+                    ref<uint8>(data_, 0) = LOGIN_ERROR_CHANGE_PASSWORD;
+                    do_write(1);
+                }
+
+                if (status & ACCOUNT_STATUS_CODE::NORMAL)
+                {
+                    // Account info verified, grab password
+                    std::string updated_password(data_ + 0x30, 16);
+
+                    if (updated_password == "")
+                    {
+                        ShowWarning(fmt::format("login_parse: Empty password; Could not update password for user <{}>.", escaped_name));
+                        ref<uint8>(data_, 0) = LOGIN_ERROR_CHANGE_PASSWORD;
+                        do_write(1);
+                        return;
+                    }
+
+                    char escaped_updated_password[16 * 2 + 1];
+                    sql->EscapeString(escaped_updated_password, updated_password.c_str());
+
+                    fmtQuery = "UPDATE accounts SET accounts.timelastmodify = NULL WHERE accounts.id = %d";
+                    sql->Query(fmtQuery, accid);
+
+                    fmtQuery = "UPDATE accounts SET accounts.password = PASSWORD('%s') WHERE accounts.id = %d";
+                    ret      = sql->Query(fmtQuery, escaped_updated_password, accid);
+                    if (ret == SQL_ERROR)
+                    {
+                        ShowWarning(fmt::format("login_parse: Error trying to update password in database for user <{}>.", escaped_name));
+                        ref<uint8>(data_, 0) = LOGIN_ERROR_CHANGE_PASSWORD;
+                        do_write(1);
+                        return;
+                    }
+
+                    memset(data_, 0, 33);
+                    ref<uint8>(data_, 0) = LOGIN_SUCCESS_CHANGE_PASSWORD;
+                    do_write(33);
+
+                    ShowInfo("login_parse: password updated successfully.");
+                    return;
+                }
+            }
+            break;
+            default:
+            {
+                ShowError(fmt::format("Unhandled auth code: {}", code));
+            }
+            break;
+        }
+    }
+
+    void write_func() override
+    {
+        do_read();
+    }
+
+    void handle_error(std::error_code ec, std::shared_ptr<handler_session> self) override
```

Comment:
Should this log?

Reply:
Logging errors here is very noisy. I've disabled it due to the fact that any early disconnection logs. It would be quite spammy with little info of use. I've added a comment to reflect as such.